mne/io/nsx/nsx.py

```diff
@@ -0,0 +1,601 @@
+                description="BAD_ACQ_SKIP",
+                orig_time=self.info["meas_date"],
+            )
+            self.set_annotations(annot)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a chunk of raw data."""
+        dtype = self._raw_extras[fi]["orig_format"]
+        first_samps = self._raw_extras[fi]["timestamp"]
+        recording_extents = self._raw_extras[fi]["nb_data_points"]
+        offsets = self._raw_extras[fi]["offset_to_data_block"]
+        for first_samp, recording_extent, offset in zip(
+            first_samps, recording_extents, offsets
+        ):
+            if start > first_samp + recording_extent:
+                # There is nothing to read in this chunk
+                continue
+            if stop < first_samp:
+                # There is nothing to read in this chunk
+                continue
+            i_start = max(start, first_samp)
+            i_stop = min(stop, first_samp + recording_extent)
+            _read_segments_file(
+                self,
+                data[:, i_start - start : i_stop - start],
+                idx,
+                fi,
+                i_start - first_samp,
+                i_stop - first_samp,
+                cals,
+                mult,
+                dtype,
+                n_channels=None,
+                offset=offset,
+                trigger_ch=None,
+            )
+
+
+def _get_file_size(filename):
+    """Return the file size in bytes for the given file."""
+    with open(filename, "rb") as fid:
+        fid.seek(0, os.SEEK_END)
+        file_size = fid.tell()
+    return file_size
+
+
+def _read_header(fname):
+    nsx_file_id = np.fromfile(fname, count=1, dtype=[("file_id", "S8")])[0][
+        "file_id"
+    ].decode()
+
+    if nsx_file_id in ["NEURALCD", "BRSMPGRP"]:
+        basic_header = _read_header_22_and_above(fname)
+    elif nsx_file_id == "NEURALSG":
+        raise NotImplementedError(
+            "NSx file id (= NEURALSG), i.e., file"
+            "version 2.1 is currently not supported."
+        )
+    else:
+        raise ValueError(
+            f"NSx file id (={nsx_file_id}) does not match"
+            "with supported file ids:"
+            "('NEURALCD', 'BRSMPGRP')"
+        )
+
+    try:
+        time_origin = datetime(
+            *[
+                basic_header[xx]
+                for xx in (
+                    "year",
+                    "month",
+                    "day",
+                    "hour",
+                    "minute",
+                    "second",
+                    "millisecond",
+                )
+            ],
+            tzinfo=timezone.utc,
+        )
+        [
+            basic_header.pop(xx)
+            for xx in (
+                "year",
+                "month",
+                "day",
+                "hour",
+                "minute",
+                "second",
+                "millisecond",
+            )
+        ]
```

Comment:
Instead of this two-step process, why not just `.pop` inside of the `datetime` constructor like
```suggestion
        time_origin = datetime(
            *[
                basic_header.pop(xx)
                for xx in (
                    "year",
                    "month",
                    "day",
                    "hour",
                    "minute",
                    "second",
                    "millisecond",
                )
            ],
            tzinfo=timezone.utc,
        )
```
I think the outcome will be the same either way